### ISSUE TITLE
Java::JavaLang::IllegalAccessError (tried to access method org.postgresql.jdbc.PgResultSetMetaData.getField(I)Lorg/postgresql/core/Field; from class org.postgresql.jdbc.PgResultSetMetaDataWrapper)

### ISSUE THREAD

**Issue by mengu** [jruby/activerecord-jdbc-adapter#965]
hello everyone

i've been trying to use tomcat's connection pooling feature and i get the error in the title. 

**rails version:** 5.2.0
**jruby version:** 9.2.5.0 (and 9.2.0.0)
**ar-jdbc-adapter version:** master
**tomcat version:** 9.0.13 (also tried with glassfish 5 - build 25)
**postgresql jar version:** 42.1.4 (also tried with 42.2.5)

here are the steps i've followed:

- changed config/database.yml to the following:
```
production:
  adapter: postgresql
  jndi: jdbc/myJndiSource
```

(i have also tried setting jndi to `java:comp/env/jbc/myJndiSource`)

- configured my web.xml and tomcat's context.xml accordingly.
- added postgresql jar file to lib dirs of tomcat and glassfish. 

result is the following stack trace:
```
  Java::JavaLang::IllegalAccessError (tried to access method org.postgresql.jdbc.PgResultSetMetaData.getField(I)Lorg/postgresql/core/Field; from class org.postgresql.jdbc.PgResultSetMetaDataWrapper):
org.postgresql.jdbc.PgResultSetMetaDataWrapper.getField(PgResultSetMetaDataWrapper.java:21)
arjdbc.postgresql.PostgreSQLResult.columnTypeMap(PostgreSQLResult.java:81)
arjdbc.jdbc.JdbcResult.toARResult(JdbcResult.java:128)
arjdbc.postgresql.PostgreSQLResult.toARResult(PostgreSQLResult.java:171)
arjdbc.postgresql.PostgreSQLRubyJdbcConnection.mapQueryResult(PostgreSQLRubyJdbcConnection.java:282)
arjdbc.jdbc.RubyJdbcConnection$15.call(RubyJdbcConnection.java:1100)
arjdbc.jdbc.RubyJdbcConnection$15.call(RubyJdbcConnection.java:1091)
arjdbc.jdbc.RubyJdbcConnection.withConnection(RubyJdbcConnection.java:3438)
arjdbc.jdbc.RubyJdbcConnection.withConnection(RubyJdbcConnection.java:3404)
arjdbc.jdbc.RubyJdbcConnection.execute_query(RubyJdbcConnection.java:1091)
arjdbc.jdbc.RubyJdbcConnection$INVOKER$i$1$0$execute_query.call(RubyJdbcConnection$INVOKER$i$1$0$execute_query.gen)
org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:346)
org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:172)
org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:317)
org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72)
```

it looks like the problem is caused by [this line](https://github.com/jruby/activerecord-jdbc-adapter/blob/ba81e827014dee4765812f58c3fb66556e2193f2/src/java/arjdbc/postgresql/PostgreSQLResult.java#L81) of PostgreSQL.java file and it looks like getField function is protected: [https://jdbc.postgresql.org/documentation/publicapi/org/postgresql/jdbc/PgResultSetMetaData.html#getField-int-](https://jdbc.postgresql.org/documentation/publicapi/org/postgresql/jdbc/PgResultSetMetaData.html#getField-int-)

**Comment by mengu** [comment]
so i think i've figured out what exactly is going on.

1. [this class](https://github.com/jruby/activerecord-jdbc-adapter/blob/97f84e23add067f82d7fadb6e5c8347e73db9ca3/src/java/arjdbc/postgresql/PgResultSetMetaDataWrapper.java) exists to access protected method getField.
2. it works without any problems in the console or standalone usage because jruby successfully generates the jvm class files for it.
3. i've just copied jruby's postgresql-42.1.4.jar file to $CATALINA_HOME/lib directory.
4. tomcat has no idea this jruby generated class exists hence the exception.

so i believe the question now is how do i get a jar for this jdbc-postgresql-adapter repository and make it available to tomcat as well?

**Comment by davecramer** [comment]
Why do you need access to getField ?

**Comment by headius** [comment]
...my previous comment also means that having getFieldOid and getFieldMod might be acceptable adds if you don't want to expose the Field objects directly.

**Comment by davecramer** [comment]
seems to me that getColumnTypeName should have been used ?

**Comment by rdubya** [comment]
Hi @davecramer, we need the old and mod because that is how the ORM keys the Hash of type converters that convert the db types to ruby types. I don't think there is a way for us to work around that using just the type because we need to things like what precision has been set on that column (as an example)

**Comment by kares** [comment]
should be resolved in >= **52.3** ... with https://github.com/jruby/activerecord-jdbc-adapter/commit/87b6e5b03d62411061d5257f670222a72fdaa2fe#diff-77209b33917121d0485cc89e1f1e6478
